app/controllers/registrations_controller.rb

```diff
@@ -628,21 +628,23 @@ def payment_completion
       #   this behavior differs and we overwrite created_at manually, see #stripe_webhook above.
     end
 
-    # Payment Intent lifecycle as per https://stripe.com/docs/payments/intents#intent-statuses
-    case stored_intent.payment_record.stripe_status
+    # For details on what the individual statuses mean, please refer to the comments
+    #   of the `enum :wca_status` declared in the `payment_intent.rb` model
+    case stored_intent.wca_status
     when 'succeeded'
       flash[:success] = t("registrations.payment_form.payment_successful")
-    when 'requires_action'
-      # Customer did not complete the payment
-      # For example, 3DSecure could still be pending.
+    when 'pending'
       flash[:warning] = t("registrations.payment_form.errors.payment_pending")
-    when 'requires_payment_method'
-      # Payment failed. If a payment fails, it is "reset" by Stripe,
-      # so from our end it looks like it never even started (i.e. the customer didn't choose a payment method yet)
+    when 'created'
```

Comment:
I don't fully follow why a `created` status should mean that we display "The payment was rejected by Stripe" - I understand this is just porting what the old system was doing, just want to confirm that the same logic should still apply? 

ie, are there reasons besides the payment being rejected that we might end up in the `created` wca_status?

Reply:
We map to wca `created` only from Stripe status `requires_payment_method`. And that Stripe status is documented as follows:

> If the payment attempt fails (for example due to a decline), the PaymentIntent’s status returns to requires_payment_method so that the payment can be retried.

See the bottom half of https://docs.stripe.com/payments/paymentintents/lifecycle

In essence, `created` means "has been initiated on the payment provider's side **but nothing else happened with it yet**". In particular, even the user trying to interact with the PI by entering an expired credit card, or waiting for funds to be transferred, would count as "some interaction" with the PI, and would thus move the lifecycle forward. `created` really means _the very initial state_ and that can only be reached if a PI has been reset.

Reply:
Makes sense, thanks! Still need to do some basic testing before approving, should get to it this evening